src/Tracer.php

```diff
@@ -186,22 +203,22 @@ private function methodCode(TestMethod $testMethod): string
         $start = $reflectionMethod->getStartLine() - 1 + 2;
         $length = $reflectionMethod->getEndLine() - 1 - $start;
 
-        $codeLines = \array_slice(
-            \file($reflectionMethod->getFileName()),
-            $start,
-            $length,
-        );
+        $testFileName = $reflectionMethod->getFileName();
+        assert(is_string($testFileName));
```

Comment:
I do understand why this is added. But I don't understand, why the `assert()` is not required in the two lines above, where `false` is also part of the return type.

Reply:
I do not know how PHPStan figure it either. Looks like this is somehow linked to phpdoc typing, because when I treat them as uncertain it says that this is not only a string.